docs/api.md

```diff
@@ -1887,6 +1888,11 @@ The method evluates the XPath expression relative to the elementHandle. If there
 
 This method returns the bounding box of the element (relative to the main frame), or `null` if the element is not visible.
 
+#### elementHandle.boxModel()
+- returns: <[Promise]<?[Object]>>
+
+This method returns the [box model](https://chromedevtools.github.io/devtools-protocol/tot/DOM/#type-BoxModel) for the given element, or `null` if the element is not visible.
```

Comment:
the protocol quads are a pain to work with; can we have an array of {x, y} objects instead?

A simple example to get a center of a quad would be beneficial too:

```js
const cx = (border[0].x + border[1].x + border[2].x + border[3].x) / 4;
const cy = (border[0].y + border[1].y + border[2].y + border[3].y) / 4;
```

Reply:
My idea was to return full original model.
We can place calculation examples into readme, right?

Or should I calculate and return quads as in `boundingBox` method?

Reply:
> My idea was to return full original model.

@schelkun I see. I don't like the protocol's output because it's cryptic. I'd rather have the boxModel to return arrays of x/y objects:

```js
const boxModel = await handle.boxModel();
boxModel.border; // Array<!{x: number, y: number}>
boxModel.content; // Array<!{x: number, y: number}>
boxModel.margin; // Array<!{x: number, y: number}>
boxModel.padding; // Array<!{x: number, y: number}>
```

> We can place calculation examples into readme, right?

Yes, that's the right place